azure/services/roleassignments/client.go

```diff
@@ -51,18 +60,52 @@ func newRoleAssignmentClient(subscriptionID string, baseURI string, authorizer a
 	return roleClient
 }
 
-// Create creates a role assignment.
-// Parameters:
-// scope - the scope of the role assignment to create. The scope can be any REST resource instance. For
-// example, use '/subscriptions/{subscription-id}/' for a subscription,
-// '/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}' for a resource group, and
-// '/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/providers/{resource-provider}/{resource-type}/{resource-name}'
-// for a resource.
-// roleAssignmentName - the name of the role assignment to create. It can be any valid GUID.
-// parameters - parameters for the role assignment.
-func (ac *azureClient) Create(ctx context.Context, scope string, roleAssignmentName string, parameters authorization.RoleAssignmentCreateParameters) (authorization.RoleAssignment, error) {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "roleassignments.AzureClient.Create")
+// Get gets the specified role assignment by the role assignment name.
+func (ac *azureClient) Get(ctx context.Context, spec azure.ResourceSpecGetter) (interface{}, error) {
+	ctx, span := tele.Tracer().Start(ctx, "roleassignments.AzureClient.Get")
+	defer span.End()
+	roleAssignmentSpec := spec.(*RoleAssignmentSpec)
```

Comment:
@Jont828 -- In this `Get` `scope` is required. I had to cast here to get the scope. 

On a different note -- this get was previously not implemented and as of now we can keep it as a no-op by returning `nil, nil` until we do some more modification to async framework to fetch arbitrary fields. 

Or we can keep as it is now -- and change once the async framework have such provision of fetching arbitrary fields.

Reply:
Yeah I don't think we can implement Get() currently without casting to get the scope. I think we can leave it as is now since it's used in the async code to see if an existing resource is there. Otherwise if it's a no-op, Parameters() would always get `existing == nil` and we'd be making a CreateOrUpdateAsync() call everytime. I think it would make sense to address this in a PR to add to the ResourceSpecGetter interface. Any thoughts @CecileRobertMichon?

Reply:
+1 to implementing this GET so we don't do a PUT for every Reconcile(). 

Make sure you handle the error when casting and return an error if `spec` is not a `RoleAssignmentSpec`.